clang/lib/CodeGen/CGPointerAuth.cpp

```diff
@@ -0,0 +1,77 @@
+//===--- CGPointerAuth.cpp - IR generation for pointer authentication -----===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file contains common routines relating to the emission of
+// pointer authentication operations.
+//
+//===----------------------------------------------------------------------===//
+
+#include "CGCXXABI.h"
```

Comment:
It looks like that only two includes are needed: CodeGenModule.h and clang/CodeGen/CodeGenABITypes.h. Maybe some other includes would become needed in future, but let's add them in corresponding PRs, not here.